src/XMakeCommandLine/UnitTests/XMake_Tests.cs

```diff
@@ -699,7 +698,22 @@ private void RobustDelete(string path)
             }
         }
 
+        /// <summary>
+        /// Run msbuild.exe within corerun.exe
+        /// It concatenates the given msbuild path to the given parameters and sends it to corerun.
+        /// </summary>
+        private string ExecMSBuild(string pathToMsBuildExe, string msbuildParameters, bool expectSuccess = true)
+        {
 #if FEATURE_RUN_EXE_IN_TESTS
```

Comment:
Shouldn't it be invisible / indifferent for the method's callers how msbuild is called? In which case ExecMSBuild has the contract to "somehow" end up invoking msbuild.exe with the given parameters; how it does that should not matter.